src/ert/ensemble_evaluator/evaluator.py

```diff
@@ -540,6 +552,65 @@ def _get_ens_id(source: str) -> str:
     async def wait_for_evaluation_result(self) -> bool:
         return await self._evaluation_result
 
+    async def evaluate(
+        self,
+    ) -> None:
+        """
+        This method does the actual work of evaluating the ensemble. It
+        prepares and executes the necessary bookkeeping, prepares and executes
+        the driver and scheduler, and dispatches pertinent events.
+
+        Before returning, it always dispatches an Event describing
+        the final result of executing all its jobs through a scheduler and driver.
+        """
+
+        if not self.ensemble.id_:
+            raise ValueError("Ensemble id not set")
+        if not self._config:
+            raise ValueError("no config")  # mypy
+        try:
+            await self.send_event(EnsembleStarted(ensemble=self.ensemble.id_))
+
+            min_required_realizations = (
+                self.ensemble.min_required_realizations
+                if self.ensemble._queue_config.stop_long_running
+                else 0
+            )
+
+            self._scheduler.add_dispatch_information_to_jobs_file()
+            scheduler_finished_successfully = await self._scheduler.execute(
+                min_required_realizations
+            )
+        except PermissionError as error:
+            logger.exception(f"Unexpected exception in ensemble: \n {error!s}")
+            await self.send_event(EnsembleFailed(ensemble=self.ensemble.id_))
+            return
+        except Exception as exc:
+            logger.exception(
+                (
+                    "Unexpected exception in ensemble: \n".join(
+                        traceback.format_exception(None, exc, exc.__traceback__)
+                    )
+                ),
+            )
+            await self.send_event(EnsembleFailed(ensemble=self.ensemble.id_))
+            return
+        except asyncio.CancelledError:
+            print("Cancelling evaluator task!")
+            return
+        # Dispatch final result from evaluator - SUCCEEDED or CANCELLED
+        if scheduler_finished_successfully:
+            await self.send_event(EnsembleSucceeded(ensemble=self.ensemble.id_))
+        else:
+            await self.send_event(EnsembleCancelled(ensemble=self.ensemble.id_))
+
+    async def send_event(self, event: EnsembleEvent) -> None:
+        await self._events.put(event)
+
+    async def cancel(self) -> None:
```

Comment:
we can can directly: `await self._scheduler.kill_all_jobs()`; ie. this function can be removed.